packages/ui/src/machines/authenticator/actions.ts

```diff
@@ -155,18 +155,14 @@ export const setUnverifiedContactMethods = assign({
 });
 
 export const setUser = assign({
-  user: (_, event: AuthEvent) => {
+  user: async (_, event: AuthEvent) => {
     groupLog('+++setUser', 'event.data', event.data);
 
-    /**
-     * @migration Cannot be called if unauthenticated. Maybe try/catch?
-     */
-    // const user = await Auth.getCurrentUser();
-    /**
-     * @migration event.data was the fallback here,
-     *  setting the entire event.data as user
-     */
-    return { ...event.data };
+    if (event.data.isSignedIn) {
+      return await Auth.getCurrentUser();
+    } else {
+      return { ...event.data };
+    }
```

Comment:
Makes sense to me. Have we run into the case where this was called when the user was already signed in?

Reply:
Yep, setUser is called quite frequently and getCurrentUser errors out if user is already signed in.

Reply:
Discussed offline, this is not needed for now